### PR TITLE
Added bounds to SciPy optimizers.

### PR DESCRIPTION
**Pull request by mstechly**

### CODE REVIEW

File: src/python/zquantum/optimizers/scipy_optimizer.py [hunk 1]
```diff
@@ -17,24 +17,34 @@ def __init__(
         self,
         method: str,
         constraints: Optional[Tuple[Dict[str, Callable]]] = None,
+        bounds: Union[
+            scipy.optimize.Bounds,
+            Sequence[Tuple[float, float]],
+            None,
+        ] = None,
```
Comment (jamesclark-Zapata) [inline src/python/zquantum/optimizers/scipy_optimizer.py:24]: These other parameters use `Optional`, what do you think about using that here too? It adds a bit of extra text
```suggestion
        bounds: Optional[
            Union[
                scipy.optimize.Bounds,
                Sequence[Tuple[float, float]],
            ]
        ] = None,

Comment (mstechly) [reply src/python/zquantum/optimizers/scipy_optimizer.py:24]: That was the first thing I tried.
But it turns out that you can't actually use `Optional` on `Union`, as `Optional` is just a shorthand for `Union[X, None]` and it throws an error. See [SO thread](https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint).

Comment (jamesclark-Zapata) [reply src/python/zquantum/optimizers/scipy_optimizer.py:24]: TIL that doesn't work.

**Review by jamesclark-Zapata — APPROVED** [review]
Changes look good.

But `mypy` is failing (also on `main`)

**Comment by mstechly** [comment]
@jamesclark-Zapata yeah, I've noticed :/ 
Investigating...

**Pull request merged**